src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs

```diff
@@ -0,0 +1,995 @@
+            {
+                N(SyntaxKind.GlobalStatement);
+                {
+                    N(SyntaxKind.LocalDeclarationStatement);
+                    {
+                        N(SyntaxKind.VariableDeclaration);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "partial");
+                            }
+                            N(SyntaxKind.VariableDeclarator);
+                            {
+                                N(SyntaxKind.IdentifierToken, "file");
+                            }
+                        }
+                        M(SyntaxKind.SemicolonToken);
+                    }
+                }
+                N(SyntaxFacts.GetBaseTypeDeclarationKind(typeKeyword));
+                {
+                    N(typeKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        // PROTOTYPE(ft): is it fine that records parse as a single declaration here, but not other type kinds?
+        [Theory]
+        [InlineData(SyntaxKind.RecordKeyword)]
+        public void FileModifier_04(SyntaxKind typeKeyword)
+        {
+            UsingNode($$"""
+                partial file {{SyntaxFacts.GetText(typeKeyword)}} C { }
+                """);
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxFacts.GetBaseTypeDeclarationKind(typeKeyword));
+                {
+                    N(SyntaxKind.PartialKeyword);
+                    N(SyntaxKind.FileKeyword);
+                    N(typeKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void FileModifier_05()
+        {
+            UsingNode($$"""
+                file partial record struct C { }
+                """);
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.RecordStructDeclaration);
+                {
+                    N(SyntaxKind.FileKeyword);
+                    N(SyntaxKind.PartialKeyword);
+                    N(SyntaxKind.RecordKeyword);
+                    N(SyntaxKind.StructKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        // PROTOTYPE(ft): is it fine that records parse here, but not other type kinds?
```

Comment:
>is it fine that records parse here, but not other type kinds? [](http://example.com/codeflow?start=26&length=61)

I don't understand this comment. Aren't the other types parsed successfully in `FileModifier_03`?